compiler/bindings/python/iree/compiler/tools/import_onnx/__main__.py

```diff
@@ -37,19 +46,213 @@
+                ftype,
+                ip=InsertionPoint(body),
+                visibility="private" if private else None,
+            )
+            block = func_op.add_entry_block(
+                [Location.name(k) for k in graph_info.input_map.keys()]
+            )
+        imp = IREENodeImporter(
+            graph_info,
+            parent_op=func_op,
+            block=block,
+            context_cache=cc,
+            module_op=module_op,
+            module_cache=mc,
+            max_numel=max_numel,
+        )
+        for node_name, input_value in zip(graph_info.input_map.keys(), block.arguments):
+            imp._nv_map[node_name] = input_value
+        imp._populate_graph_attrs(func_op)
+        return imp
+
+    def import_initializer(
+        self, initializer: onnx.TensorProto, extern_name: Optional[str] = None
+    ) -> Value:
+        # If an explicitly specified name is given, use that; otherwise, pick
+        # up the name from the tensor proto itself
+        iname = extern_name if extern_name else initializer.name
+        dims = list(initializer.dims)
+        acc = 1
+        for d in dims:
+            acc = acc * d
+        if acc < self.max_numel:
+            imported_tensor = super().import_initializer(initializer)
+            self._nv_map[iname] = imported_tensor
+            return imported_tensor
+
+        x, t = self.create_tensor_global(initializer)
+        vtensor_type = self._cc.get_vtensor_type(
+            tuple(initializer.dims), self._cc.tensor_element_type(initializer.data_type)
+        )
+
+        with InsertionPoint(self._b), Location.name(iname):
+            old_op = util.GlobalLoadOp(t, x)
+            converted_value = Operation.create(
+                "torch_c.from_builtin_tensor",
+                results=[vtensor_type],
+                operands=[old_op.result],
+            ).result
+
+        self._nv_map[iname] = converted_value
+        tensor_as_array = numpy_helper.to_array(initializer)
+        self.param_archive.add_buffer(x, tensor_as_array)
+        return converted_value
+
+
 def main(args: argparse.Namespace):
     model_proto = load_onnx_model(args)
     context = Context()
     model_info = onnx_importer.ModelInfo(model_proto)
     m = model_info.create_module(context=context).operation
-    imp = onnx_importer.NodeImporter.define_function(model_info.main_graph, m)
+
+    imp: Any = None
+    if args.externalize_params:
+        imp = IREENodeImporter.define_function(model_info.main_graph, m, args.max_numel)
+    else:
+        if args.max_numel:
+            warnings.warn(
+                "'--max-numel' has no effect until externalization is enabled with '--externalize-params'"
+            )
+        imp = onnx_importer.NodeImporter.define_function(model_info.main_graph, m)
     imp.import_all()
+
     if not args.no_verify:
         m.verify()
 
+    if args.externalize_params:
+        imp.param_archive.create_archive_file(args.save_params_to)
```

Comment:
I'm a bit wary of a default `args.save_params_to` since running multiple models through the importer will replace the params file. Let's make `save_params_to` not take a default, then do something like:

```python
if args.externalize_params:
    default_param_path = Path(args.output_file).parent / Path(args.output_file).stem + "_params.irpa"
    param_path = default_param_path if args.params_save_to is None else args.params_save_to
    imp.param_archive.create_archive_file(param_path)
```

This way, the params will be located in the same directory as the output file by default, and will have the name be `<name of output file>_params.irpa`.